### DISCUSSION TITLE
How to store HLS streams recordings to AWS S3 bucket?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5090] [FAQs] [answered]
I would like to use an AWS S3 bucket to store the stream recordings as HLS files.
It appears as AMS uploads the HLS files to the bucket only after the stream ends. Is it possible to upload these to the S3 bucket in real time as they get created?

Also, how can I make sure that the HLS files do not get overwritten if using the same streamId?

Thank you

**Comment by yashtandon113** [comment, marked as answer]
Thank you for asking this question.
There are two methods for uploading recordings from AMS to the S3 bucket.

1. You can integrate the S3 bucket with AMS, and recordings will be uploaded whenever the stream is finished. For additional information, see the [s3 integration document](https://antmedia.io/docs/guides/recording-live-streams/s3-integration-http-forwarding/).

2. Use s3 fuse to upload the files to the bucket at the same time.

More information about the s3 fuse can be found at https://github.com/s3fs-fuse/s3fs-fuse.
You can also immediately mount your application's streams folder to your S3 bucket by following the instructions below:

- Install s3fs

 `sudo apt install s3fs`

- Add the credentials

```
echo ACCESS_KEY_ID:SECRET_ACCESS_KEY > /etc/passwd-s3fs
chmod 600 /etc/passwd-s3fs
```

- Mount streams folder of your application to the s3 bucket by creating the mount point.

```
sudo s3fs -o dbglevel=info -o curldbg -o allow_other -o default_acl='public-read' -o use_cache=/tmp/s3-cache <your-bucket-name> /usr/local/antmedia/webapps/App-Name/streams
```

- Once the mount point has been added successfully, run the **df -h** command and look for the mount point in an output. It should be as follows:

 `s3fs    256T     0  256T   0% /usr/local/antmedia/webapps/WebRTCAppEE/streams`
 
 If you do not see this mount point in the output then there is some issue with the configuration.
